### ISSUE TITLE
Get original VaadinIcon constant during runtime

### ISSUE THREAD

**Issue by masbaehr** [vaadin/vaadin-icons-flow#43]
I'm looking for a better way to replace this ugly workaround
```java
if(ib.getIcon() != null){
					String iconAttribute = ib.getIcon().getElement().getAttribute("icon");
					b.setIcon(new Icon(VaadinIcon.valueOf(iconAttribute.toUpperCase().replace("VAADIN:", "").replace("-", "_"))));
				}
```
For a better understanding: I'm dynamically creating new buttons from previous button instances. A simple call like the following is not possible, as it would remove the Icon from the _oldButton_ instance.
```java
newButton.setIcon(oldButton.getIcon());
```
So something like "getVaadinIcon()" inside the Icon.class would be very helpful. (Obviously this would be only possible when the Icon has been created using VaadinIcon - but this is probably okay.

**Comment by TatuLund** [comment]
Yes it would be prefferred to have VaadinIcon names compatible with client side. The even more common example where you need to fix the name is if you want to use icons in the template renderer like in this example:

https://github.com/TatuLund/filesystem-dataprovider-flow/blob/master/src/main/java/org/vaadin/filesystemdataprovider/FileSelect.java#L70

**Comment by TatuLund** [comment]
Interesting detail is that the same workaround is applied in internal implementation of Icon.java

https://github.com/vaadin/vaadin-icons-flow/blob/master/vaadin-icons-flow/src/main/java/com/vaadin/flow/component/icon/Icon.java#L59

**Comment by web-padawan** [comment]
This sounds like a good candidate for fixing in scope of onboarding improvements.
Maybe the internal workaround should be exposed as a public API.
